kie-maven-plugin/src/main/java/org/kie/maven/plugin/GeneratePMMLModelMojo.java

```diff
@@ -0,0 +1,322 @@
+        // GeneratePMMLModelMojo is executed when BuildMojo and GenerateModelMojo aren't and vice-versa
+        boolean modelParameterEnabled = PMMLModelMode.modelParameterEnabled(generatePMMLModel);
+        boolean modelCompilerInClassPath = isModelCompilerInClassPath(project.getDependencies());
+        if (modelParameterEnabled && modelCompilerInClassPath) {
+            generateModel();
+        } else if (modelParameterEnabled) { // !modelCompilerInClassPath
+            getLog().warn("You're trying to build rule assets in a project from an executable rule model, but you did not provide the required dependency on the project classpath.\n" +
+                                  "To enable executable rule models for your project, add the `drools-model-compiler` dependency in the `pom.xml` file of your project.\n");
+        }
+    }
+
+    private void generateModel() throws MojoExecutionException {
+        ClassLoader contextClassLoader = Thread.currentThread().getContextClassLoader();
+        try {
+            Set<URL> urls = new HashSet<>();
+            for (String element : project.getCompileClasspathElements()) {
+                urls.add(new File(element).toURI().toURL());
+            }
+
+            project.setArtifactFilter(new CumulativeScopeArtifactFilter(Arrays.asList("compile",
+                                                                                      "runtime")));
+            for (Artifact artifact : project.getArtifacts()) {
+                File file = artifact.getFile();
+                if (file != null) {
+                    urls.add(file.toURI().toURL());
+                }
+            }
+            urls.add(outputDirectory.toURI().toURL());
+
+            ClassLoader projectClassLoader = URLClassLoader.newInstance(urls.toArray(new URL[0]),
+                                                                        getClass().getClassLoader());
+            Thread.currentThread().setContextClassLoader(projectClassLoader);
+        } catch (DependencyResolutionRequiredException | MalformedURLException e) {
+            throw new RuntimeException(e);
+        }
+
+        KieServices ks = KieServices.Factory.get();
+        try {
+            // FLAG TO KNOW THAT BUILD HAS BEEN LAUNCHED BY MAVEN
+            properties.put("kie-maven-plugin-launcher", "true");
+            setSystemProperties(properties);
+
+            final KieBuilderImpl kieBuilder = (KieBuilderImpl) ks.newKieBuilder(projectDir);
+            kieBuilder.setPomModel(new ProjectPomModel(mavenSession));
+            String testPath = "src" + File.separator + "test" + File.separator + "java";
+            String pmmlResourcePath = "src" + File.separator + "main" + File.separator + "resources" + File.separator + PMMLRESOURCES;
+            kieBuilder.buildAll(ExecutableModelMavenProject.SUPPLIER,
+                                s -> !s.contains(testPath) && !s.contains(pmmlResourcePath));
```

Comment:
Not sure if `File.separator` can be used here, probably is better to use `/`.
As far as I remember KieFileSystem only supports `/` so this code could fail on Windows where `File.separator` is usually `\`.
@mariofusco Wdyt?